### PR TITLE
ProfilesCheck: new check for UnmatchedProfilePackageUnmask

### PR DESCRIPTION
**Pull request by arthurzam**
Add missing check from repoman, for unmatched unmask of atom in `package.mask` files. This checks for any unmask of package, which isn't masked in parent profiles.

I have a feeling my desc output and class documentation for the result are **very bad**, so help is needed :smile:

Matches in gentoo tree using `pkgcheck scan -k UnmatchedProfilePackageUnmask`:
```
profiles
  UnmatchedProfilePackageUnmask: 'arch/powerpc/ppc64/64le/package.mask': unmask of not masked package: 'dev-java/openjdk-bin'
  UnmatchedProfilePackageUnmask: 'arch/powerpc/ppc64/64le/package.mask': unmask of not masked package: 'net-im/telegram-desktop'
  UnmatchedProfilePackageUnmask: 'arch/riscv/rv64gc/lp64d/package.mask': unmask of not masked package: 'dev-java/openjdk'
  UnmatchedProfilePackageUnmask: 'arch/riscv/rv64gc/lp64d/package.mask': unmask of not masked package: 'media-tv/kodi'
  UnmatchedProfilePackageUnmask: 'arch/riscv/rv64gc/lp64d/package.mask': unmask of not masked package: 'virtual/jdk'
  UnmatchedProfilePackageUnmask: 'arch/riscv/rv64gc/lp64d/package.mask': unmask of not masked package: 'virtual/jre'
  UnmatchedProfilePackageUnmask: 'features/musl/package.mask': unmask of not masked package: 'sys-libs/libucontext'
  UnmatchedProfilePackageUnmask: 'features/musl/package.mask': unmask of not masked package: 'sys-libs/musl'
  UnmatchedProfilePackageUnmask: 'features/musl/package.mask': unmask of not masked package: 'virtual/libcrypt:0/1'
  UnmatchedProfilePackageUnmask: 'prefix/darwin/macos/features/fsf-gcc-nold/package.mask': unmask of not masked package: 'sys-devel/native-cctools'
```

Resolves: https://github.com/pkgcore/pkgcheck/issues/369

### CODE REVIEW

File: src/pkgcheck/checks/profiles.py [hunk 1]
```diff
@@ -28,6 +28,20 @@ def desc(self):
         return f'{self.path!r}: unknown package: {self.atom!r}'
 
 
+class UnmatchedProfilePackageUnmask(results.ProfilesResult, results.Warning):
+    """Profile files includes package entry unmask that isn't masked in the
```
Comment (thesamesam) [inline src/pkgcheck/checks/profiles.py:32]: "The profile's files include a package.unmask (or similar) entry which negates a non-existent mask, i.e. it undoes a mask which doesn't exist in the parent profile."

File: src/pkgcheck/checks/profiles.py [hunk 2]
```diff
@@ -28,6 +28,23 @@ def desc(self):
...
+    No atoms matching this entry were found in the parent profile to unmask."""
+
+    def __init__(self, path, atom):
+        super().__init__()
+        self.path = path
+        self.atom = str(atom)
+
+    @property
+    def desc(self):
+        return f'{self.path!r}: unmask of not masked package: {self.atom!r}'
```
Comment (arthurzam) [inline src/pkgcheck/checks/profiles.py:45]: @thesamesam does this look fine?

Comment (arthurzam) [reply src/pkgcheck/checks/profiles.py:45]: OK, then lets merge and release

Comment (thesamesam) [reply src/pkgcheck/checks/profiles.py:45]: sounds good!

**Pull request merged**